### PR TITLE
infoschema: migrate `dataForProcesslist` from package `infoschema` to `executor`

### LINKED ISSUE #15020: UCP: Migrate `dataForProcesslist` from package `infoschema` to `executor`
## Description

This is sub-task of https://github.com/pingcap/tidb/issues/14828

The package `infoschema` should be used to store meta-information for virtual tables only.
But there is some retrieval data logic of the virtual table, which messes the `infoschema` .
PR #14704 has begun to migrate them. You can reference #14704 to help you migrate the `dataForProcesslist`.

## Score 

* 50

## Mentor(s)

* @lonng 

## Recommended Skills

* Go

## Learning Materials

- PR: #14704

### PR DESCRIPTION
**Pull request by hsqlu**
UCP: #15020 

### What problem does this PR solve? 
fixes #15020 

### What is changed and how it works?
move `dataForProcesslist` from infoschema to executor.
move `dataForClusterProcesslist` from infoschema to executor.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

### CODE REVIEW

File: infoschema/tables_test.go [hunk 1]
```diff
@@ -743,76 +740,6 @@ func (s *testTableSuite) TestForTableTiFlashReplica(c *C) {
 	tk.MustQuery("select TABLE_SCHEMA,TABLE_NAME,REPLICA_COUNT,LOCATION_LABELS,AVAILABLE, PROGRESS from information_schema.tiflash_replica").Check(testkit.Rows("test t 2 a,b 1 1"))
 }
 
-func (s *testClusterTableSuite) TestForClusterServerInfo(c *C) {
```
Comment (lonng) [inline infoschema/tables_test.go]: Why remove the test cases?

Comment (hsqlu) [reply infoschema/tables_test.go]: By mistake, sorry!  Just wondering why this unit test failed when I run `go test` under this package `infoshema`.

**Comment by reafans** [comment]
Hi, @hsqlu, thanks for your contribution, please fix the lint below:
```
[2020-03-19T20:27:57.847Z] # github.com/pingcap/tidb/infoschema

[2020-03-19T20:27:57.847Z] infoschema/tables.go:35:2: imported and not used: "github.com/pingcap/tidb/privilege"

[2020-03-19T20:27:57.847Z] make: *** [importer] Error 2

[2020-03-19T20:27:58.105Z] # github.com/pingcap/tidb/infoschema

[2020-03-19T20:27:58.105Z] infoschema/tables.go:35:2: imported and not used: "github.com/pingcap/tidb/privilege"

[2020-03-19T20:27:58.105Z] # github.com/pingcap/tidb/infoschema

[2020-03-19T20:27:58.105Z] infoschema/tables.go:35:2: imported and not used: "github.com/pingcap/tidb/privilege"

[2020-03-19T20:27:58.105Z] make: *** [server] Error 2

[2020-03-19T20:27:58.105Z] make: *** [server] Error 2

script returned exit code 2
```

**Comment by hsqlu** [comment]
> Hi, @hsqlu, thanks for your contribution, please fix the lint below:
> 
> ```
> [2020-03-19T20:27:57.847Z] # github.com/pingcap/tidb/infoschema
> 
> [2020-03-19T20:27:57.847Z] infoschema/tables.go:35:2: imported and not used: "github.com/pingcap/tidb/privilege"
> 
> [2020-03-19T20:27:57.847Z] make: *** [importer] Error 2
> 
> [2020-03-19T20:27:58.105Z] # github.com/pingcap/tidb/infoschema
> 
> [2020-03-19T20:27:58.105Z] infoschema/tables.go:35:2: imported and not used: "github.com/pingcap/tidb/privilege"
> 
> [2020-03-19T20:27:58.105Z] # github.com/pingcap/tidb/infoschema
> 
> [2020-03-19T20:27:58.105Z] infoschema/tables.go:35:2: imported and not used: "github.com/pingcap/tidb/privilege"
> 
> [2020-03-19T20:27:58.105Z] make: *** [server] Error 2
> 
> [2020-03-19T20:27:58.105Z] make: *** [server] Error 2
> 
> script returned exit code 2
> ```

Resloved. PTAL Thanks!

**Pull request merged**